tests/tests/test_monitor_client.py

```diff
@@ -1187,10 +1189,84 @@ def test_monitorclient_logs_and_services(self, monitor_commercial_setup_no_clien
         assert fds_count1.rstrip().isnumeric()
         fds_count1 = int(fds_count1.rstrip())
         logger.info(
-            "test_monitorclient_alert_email: currently %s fds open" % fds_count0
+            "test_monitorclient_logs_and_services: currently %s fds open" % fds_count0
         )
         assert abs(fds_count1 - fds_count0) <= max_fds_count_diff
-        logger.info("test_monitorclient_alert_email: fds count have not increased")
+        logger.info(
+            "test_monitorclient_logs_and_services: fds count have not increased"
+        )
+
+    def test_monitorclient_logs_and_patterns(self, monitor_commercial_setup_no_client):
+        """Tests the monitor client email alerting for multiple patterns types"""
+        mailbox_path = "/var/spool/mail/local"
+        wait_for_alert_interval_s = 8
+        expected_from = "noreply@mender.io"
+        user_name = "some.user+{}@example.com".format(str(uuid.uuid4()))
+        devid, _, auth, mender_device = self.prepare_env(
+            monitor_commercial_setup_no_client, user_name
+        )
+        inventory = Inventory(auth)
+        logger.info("test_monitorclient_logs_and_patterns: env ready.")
+
+        logger.info(
+            "test_monitorclient_logs_and_patterns: email alert on log file containing a pattern scenario."
+        )
+        log_file = "/tmp/mylog.log"
+        log_pattern = "session opened for user \\w+"
+        service_name = "mylog"
+
+        mender_device.run("echo 'some line 1' >> " + log_file)
+        mender_device.run("echo 'some line 2' >> " + log_file)
+        prepare_log_monitoring(
+            mender_device,
+            service_name + "-pcre",
+            "@tail -f " + log_file,
+            log_pattern,
+            use_ctl=True,
+        )
+        mender_device.run("systemctl restart mender-monitor")
+        time.sleep(2 * wait_for_alert_interval_s)
+
+        mender_device.run(
+            "echo -ne 'a new session opened for user root now\nsome line 3\n' >> "
+            + log_file
+        )
+        time.sleep(wait_for_alert_interval_s)
+
+        mender_device.run("echo 'some line 4' >> " + log_file)
+        mender_device.run("echo 'some line 5' >> " + log_file)
+
+        mender_device.run(
+            "echo -ne 'a new session opened for user root now\nsome line 5\n' >> "
+            + log_file
+        )
+        time.sleep(2 * wait_for_alert_interval_s)
+
+        alerts, alert_count = self.get_alerts_and_alert_count_for_device(
+            inventory, devid
+        )
+        assert (True, 1) == (alerts, alert_count)
```

Comment:
Why won't we have two alerts? :thinking:

Reply:
the same service, check, on the same device always results in one alert as counted by inventory: this is by design, it is counted as one current critical, see the earlier test logs_and_services, where we have 2, because there are 2 services.